src/frontend-samples/serialize-view-sample/SerializeViewUI.tsx

```diff
@@ -0,0 +1,229 @@
+
+interface SerializeViewUIProps {
+  iModelName: string;
+  iModelSelector: React.ReactNode;
+}
+
+interface SerializeViewUIState {
+  viewport?: Viewport;
+  iModelViews: ViewStateWithName[];
+  iModelViewIndex: number;
+  jsonMenuVisible: boolean;
+  jsonMenuValue: string;
+  jsonError: string | undefined;
+  loadStateError: string | undefined;
+}
+
+const iModelIdToNameDict: { [id: string]: string } = {
+  "bef6d098-bcf4-41cf-a0b2-4a49553fefa2": "Metrostation",
+  "97a67f36-8efa-499c-a6ed-a8e07f38a410": "Retail Building",
+}
+
+export default class SerializeViewUI extends React.Component<SerializeViewUIProps, SerializeViewUIState> {
+
+  public state: SerializeViewUIState = {
+    iModelViews: [],
+    jsonMenuVisible: false,
+    jsonMenuValue: "",
+    iModelViewIndex: 0,
+    jsonError: "",
+    loadStateError: "",
+  }
+
+  /** Dictionary of imodelId's to array of viewstates */
+  public allSavedViews: IModelViews[] = [...sampleViewStates];
+
+  private readonly _onSaveStateClick = () => {
+    const currentimodelid = this.state.viewport?.iModel?.iModelId;
+    /** Check that the viewport is not null, and there is an iModel loaded with an ID */
+    if (this.state.viewport !== undefined && currentimodelid !== undefined) {
+
+      /** Serialize the current view */
+      const viewStateProps = SerializeViewApp.serializeCurrentViewState(this.state.viewport);
+
+      /** Add that serialized view to the list of views to select from */
+      this.setState((prevState) => (
+        { iModelViews: [...prevState.iModelViews, { name: `Saved View: ${prevState.iModelViews.length + 1}`, view: viewStateProps }] }
+      ));
+    }
+  }
+
+  /** Loads the view selected */
+  private readonly _onLoadStateClick = () => {
+    if (undefined !== this.state.viewport) {
+      const view = this.state.iModelViews[this.state.iModelViewIndex].view;
+
+      //* * Load the view state. Display error message if there is one */
+      SerializeViewApp.loadViewState(this.state.viewport, view)
+        .then(() => {
+          if (this.state.loadStateError) {
+            this.setState({ loadStateError: "" });
+          }
+        })
+        .catch(() => {
+          this.setState({ loadStateError: "Error changing view: invalid view state." });
+        });
+    }
+  }
+
+  /** Show the JSON Popup window */
+  private readonly _onShowJsonViewerClick = () => {
+    this.setState({ jsonMenuVisible: true });
+  }
+
+  /** Hide the JSON popup window */
+  private _onHideJsonViewerClick = () => {
+    this.setState({ jsonMenuVisible: false });
+  }
+
+  /** Will be triggered once when the iModel is loaded. */
+  private readonly _onIModelReady = (_iModel: IModelConnection) => {
+    IModelApp.viewManager.onViewOpen.addOnce((viewport: ScreenViewport) => {
+      /** Grab the IModel with views that match the imodel loaded. */
+      const iModelId = viewport.iModel.iModelId !== undefined ? viewport.iModel.iModelId : "";
```

Comment:
I can do this. The initial reason why I didn't was because the names returned in the view object's don't match up with the UI. e.g. 'Retail Building Sample' in the drop down selector doesn't match up with viewport.iModel.name's value of 'iModel Hub Website seed file'. If a user looked into the json for changes, they'd be confused with the name shift. After looking through the code, it looks like the iModel project name is 'Retail Building Sample' but when the remote briefcase connection returns, its name returns 'Retail Building Sample'. Not sure what i need to do to fix that.

Reply:
Was over complicating it, and found a more elegant solution.